types/constants.go

```diff
@@ -8,31 +8,43 @@ package types
 
 import (
 	"math/big"
+	"sync"
 	"time"
 
 	"github.com/NebulousLabs/Sia/build"
 )
 
+// COMPATv0.4.0 - until the first 10e3 blocks have been archived, CurrentHeight
+// is needed to keep code consistent while verifying earlier blocks.
+//
+// CurrentHeight is a global variable to help with an upcoming hardfork. When
+// the height is less than 12,000, one set of code will be run. When the height
+// is greater than 12,000, a different set of code will be run. After the
+// hardfork has passed and everyone has upgraded, a checkpoint will be created
+// and blocks below 12,000 will never be viewed again.
+var CurrentHeight BlockHeight = 0
+var CurrentHeightLock sync.Mutex
```

Comment:
almost.

The amount of code that I would have had to introduce I believe is a lot higher. Maybe I didn't look hard enough for a better solution, but the only function that gets affected is the FileContract.Tax() function. It made sense to just put the limits there, instead of having a larger architecture try to choose between two different Tax funtions.